module-new/src/components/dialog/dialog.component.tsx

```diff
@@ -73,6 +75,7 @@ export const Dialog = React.forwardRef(
       className,
       data,
       overlayClassName,
+      testId,
```

Comment:
So, testId... It's one of those testing patterns that keeps falling in and out of favour, and I think we're currently in an out cycle. The ideal at the moment is 'write your tests around what the user can see and don't clutter your data-* tags', so in this case unless there's a good reason not to the recommendation would be to use `getByRole('dialog')` as your matcher.

That said, it's almost certainly going to swing back the other way in n years, and I personally think having it as an option on the component for when one can't target it in any other way isn't a bad thing, so... -shrugs expansively-

I think my suggestion overall is to keep it but to try not to use it in your own tests as a good example?

Reply:
See also 'getByText' for the buttons etc.

Reply:
I suppose that sort of makes sense... Presumably if there's multiple buttons/inputs on screen and I want to click/type into a specific one, a testId is the only way? There's not always something else to discriminate them

Reply:
All testIds now removed in favour of more accessible selectors, with the exception of the "results" div, which is only used to test the async dialog responses and isn't part of the actual component anyway